src/include/sof/lib/alloc.h

```diff
@@ -61,99 +56,6 @@ void *_brealloc(void *ptr, uint32_t flags, uint32_t caps, size_t bytes,
 		uint32_t alignment);
 void rfree(void *ptr);
 
-#if CONFIG_DEBUG_HEAP
-
-#define rmalloc(zone, flags, caps, bytes)				\
-	({void *_ptr;							\
-	do {								\
-		_ptr = _malloc(zone, flags, caps, bytes);		\
-		if (!_ptr) {						\
-			trace_mem_error("failed to alloc 0x%x bytes caps 0x%x flags 0x%x",\
```

Comment:
The big advantage here was that the callers line number would be shown in trace, is this still the case with this update ?

Reply:
No, but there is a reason for that. I believe that ultimately we do not want to have an extra debug build. It seems that nobody used the CONFIG_DEBUG_HEAP option as the code had not be able to compile. In practice, most of the code I examined dealt with the null ptr-s returned with its own more detailed error messages. Otherwise the dsp panics on the first access to null or very small non-zero memory address and stack analysis should be done.

In general, I believe that all debugability features should be always compiled in, adding as less overhead as possible. They are useful if user can enable them at run-time if necessary, otherwise they are very rarely/~never used like here.

In this particular case, the memory traces will be switched between info/dbg level selectively, just for memory management at run-time once the trace run-time config/filtering is implemented and CONFIG_DEBUG_HEAP will be eliminated (like most of other _DEBUG_ options I hope). I decided to keep (move) the macros inside *alloc() functions for real run-time mem performance analysis of those rare cases when clients know how to proceed with allocation errors following some fallback paths.